src/nvim/CMakeLists.txt

```diff
@@ -361,17 +385,98 @@ elseif(CLANG_TSAN)
   set_property(TARGET nvim APPEND_STRING PROPERTY LINK_FLAGS "-fsanitize=thread ")
 endif()
 
-add_library(libnvim STATIC EXCLUDE_FROM_ALL ${NEOVIM_GENERATED_SOURCES}
-  ${NEOVIM_SOURCES} ${NEOVIM_HEADERS})
+add_library(libnvim STATIC EXCLUDE_FROM_ALL ${NVIM_GENERATED_FOR_SOURCES} ${NVIM_GENERATED_FOR_HEADERS}
+  ${NVIM_GENERATED_SOURCES} ${NVIM_SOURCES} ${NVIM_HEADERS})
 target_link_libraries(libnvim ${NVIM_LINK_LIBRARIES})
 set_target_properties(libnvim PROPERTIES
   POSITION_INDEPENDENT_CODE ON
   OUTPUT_NAME nvim)
 set_property(TARGET libnvim APPEND_STRING PROPERTY COMPILE_FLAGS " -DMAKE_LIB ")
 
-add_library(nvim-test MODULE EXCLUDE_FROM_ALL ${NEOVIM_GENERATED_SOURCES}
-    ${NEOVIM_SOURCES} ${UNIT_TEST_FIXTURES} ${NEOVIM_HEADERS})
+add_library(nvim-test MODULE EXCLUDE_FROM_ALL ${NVIM_GENERATED_FOR_SOURCES} ${NVIM_GENERATED_FOR_HEADERS}
+  ${NVIM_GENERATED_SOURCES} ${NVIM_SOURCES} ${UNIT_TEST_FIXTURES} ${NVIM_HEADERS})
 target_link_libraries(nvim-test ${NVIM_LINK_LIBRARIES})
 set_property(TARGET nvim-test APPEND_STRING PROPERTY COMPILE_FLAGS -DUNIT_TESTING)
 
+set(NO_SINGLE_CHECK_HEADERS
+  buffer
+  charset
+  cursor_shape
+  diff
+  digraph
+  ex_cmds
+  ex_getln
+  file_search
+  fold
+  getchar
+  hardcopy
+  if_cscope
+  if_cscope_defs
+  mark
+  mbyte
+  memfile_defs
+  memline
+  memline_defs
+  menu
+  misc2
+  move
+  msgpack_rpc/server
+  ops
+  option
+  os/shell
+  os_unix
+  os/win_defs
+  popupmnu
+  quickfix
+  regexp
+  regexp_defs
+  screen
+  search
+  sha256
+  sign_defs
+  spell
+  spellfile
+  syntax
+  syntax_defs
+  tag
+  terminal
+  tui/tui
+  ugrid
+  ui
+  ui_bridge
+  undo
+  undo_defs
+  version
+  window
+)
+foreach(hfile ${NVIM_HEADERS})
+  get_filename_component(full_d ${hfile} PATH)
+  file(RELATIVE_PATH d "${PROJECT_SOURCE_DIR}/src/nvim" "${full_d}")
+  if(${d} MATCHES "^[.][.]")
+    file(RELATIVE_PATH d "${GENERATED_DIR}" "${full_d}")
+  endif()
+  get_filename_component(r ${hfile} NAME_WE)
+  if(NOT ${d} EQUAL ".")
+    set(r "${d}/${r}")
+  endif()
+
+  if(NOT ${hfile} MATCHES "[.]c[.]h$")
+    set(tsource "${GENERATED_DIR}/${r}.test-include.c")
+    set(tresult "${GENERATED_DIR}/${r}.test-include.i")
+    string(REPLACE "/" "-" texe "test-incl-${r}")
+    write_file("${tsource}" "#include \"${hfile}\"\nint main(int argc, char **argv) { return 0; }")
+    get_preproc_output(PREPROC_OUTPUT ${tresult})
```

Comment:
@ZyX-I PREPROC_OUTPUT is not used after this line. Can this be removed or did we miss something?